### ISSUE TITLE
WITCHER: Tries to load non-existent files

### ISSUE THREAD

**Issue by wahrhaft** [xoreos/xoreos#95]
When I try to run The Witcher, I run into the following problem:
```
Detected game "The Witcher"
[  0%] Declare languages (0.00s)
Setting the language to Invalid text + Invalid voices
[  7%] Loading user game config (0.05s)
[ 15%] Setting base directory (0.07s)
[ 23%] Adding extra archive directories (0.08s)
ERROR: No such directory "data/voices"
```
My first workaround is to change in WitcherEngine::initResources():

`indexMandatoryDirectory("data/voices" , 0,  0, 4);`
to
`indexOptionalDirectory("data/voices" , 0,  0, 4);`

Now we get further:
```
Detected game "The Witcher"
[  0%] Declare languages (0.00s)
Setting the language to Invalid text + Invalid voices
[  7%] Loading user game config (0.05s)
[ 15%] Setting base directory (0.07s)
[ 23%] Adding extra archive directories (0.08s)
[ 30%] Loading main KEY (0.10s)
[ 38%] Loading the localized base KEY (0.83s)
[ 46%] Indexing extra resources (0.84s)
[ 53%] Indexing Windows-specific resources (1.05s)
[ 61%] Indexing override files (1.10s)
[ 69%] Indexing language files (Invalid text + Invalid voices) (1.10s)
ERROR: No such archive file "lang_-1.key"
```

It's not detecting the language due to more missing files.  I had to comment out the check for the M1 and M2 files in detectAudio():
```C++
const Common::UString v1 = Common::UString::format("lang_%d.key"  , langID);
const Common::UString v2 = Common::UString::format("M1_%d.key"    , langID);
const Common::UString v3 = Common::UString::format("M2_%d.key"    , langID);
const Common::UString t  = Common::UString::format("dialog_%d.tlk", langID);

if (files.contains(v1, true) /*&& files.contains(v2, true) && files.contains(v3, true)*/)
	languagesVoice.push_back((Aurora::Language) i);
```
And make M1 and M2 optional in loadLanguageFiles():
```C++
_languageResources.push_back(Common::ChangeID());
archive = Common::UString::format("M1_%d.key", LangMan.getLanguageID(langVoice));
indexOptionalArchive(archive, 101, &_languageResources.back());

_languageResources.push_back(Common::ChangeID());
archive = Common::UString::format("M2_%d.key", LangMan.getLanguageID(langVoice));
indexOptionalArchive(archive, 102, &_languageResources.back());
```
Now we can load successfully:
```
Detected game "The Witcher"
[  0%] Declare languages (0.00s)
Setting the language to English text + English voices
[  7%] Loading user game config (0.03s)
[ 15%] Setting base directory (0.05s)
[ 23%] Adding extra archive directories (0.06s)
[ 30%] Loading main KEY (0.08s)
[ 38%] Loading the localized base KEY (0.80s)
[ 46%] Indexing extra resources (0.82s)
[ 53%] Indexing Windows-specific resources (1.00s)
[ 61%] Indexing override files (1.02s)
[ 69%] Indexing language files (English text + English voices) (1.03s)
[ 76%] Registering file formats (1.95s)
[ 84%] Loading game cursors (1.95s)
[ 92%] Initializing internal game config (1.97s)
[100%] Successfully initialized the engine (1.99s)
```
I installed The Witcher from my original v1.0 CD (I got it bundled for free with a graphics card quite some time ago), and then applied all the patches to take it to v1.5.  I'm not sure if this install process results in a different set of files compared to buying the game today.  I can share a full file listing from my installation if it would be helpful, or try anything else you'd like.

**Comment by DrMcCoy** [comment]
Ah, I guess those files are part of the [premium modules](http://witcher.wikia.com/wiki/Premium_module). Seems like they aren't included with the 1.5 patch?

I have the GOG version, and there they are included.

Thanks for reporting :).
I'll apply those changes to the repo tomorrow, when I'm not tired and nearly falling out of my chair.

**Comment by DrMcCoy** [comment]
Okay, I made those resource files optional now, in 3ae076f0074ea9e13906d31938ef22931b552a2d. :)
